.github/workflows/ci.yml

```diff
@@ -0,0 +1,120 @@
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+      - staging
+    paths-ignore:
+      - "doc/**"
+
+name: Continuous Integration
+
+jobs:
+  lint:
+    name: Run linters
```

Comment:
Nitpick: It looks like only one of these jobs has a name.  We could probably drop this name for consistency?